masonry/src/passes/compose.rs

```diff
@@ -75,18 +75,10 @@ fn compose_widget(
             );
             let parent_bounding_rect = parent_state.bounding_rect;
 
-            // This could be further optimized by more tightly clipping the child bounding rect according to the clip path.
```

Comment:
This note about optimization is not present after the code was moved. Is it still a valid future thing to do?

Reply:
Or is that the new call to `overlaps`? Either way, the logic in this code has changed, so the PR title that this is just adding documentation starts to become misleading.

Reply:
Ah, you renamed it after I started looking at it.

Reply:
> This note about optimization is not present after the code was moved. Is it still a valid future thing to do?

I'd have to ask @Philipp-M. I'm not sure the note about optimization made sense in the original PR. I'd rather remove it unless we can point at a specific optimization we expect to do.

Reply:
Well since the bounds might need to be bigger when transforming the AABB (due to not being able to tighly fit a transformed AABB), it's probably larger afterwards.
An optimization might be an OOB to more tightly represent the widget, or more optimally an arbitrary shape.
I'm not sure how useful that comment is, but at least it hints that this is not yet optimal.

Reply:
I'm fine not hinting at a future optimization, because I highly doubt there will be one. Tightly clipping an arbitrary path in an oriented bounding box is a hard problem, and it's unclear that it would have any performance benefit at all.

99.9% of cases are going to be AABBs anyway, for which the current code is close to optimal.

Reply:
Yeah I'm not too attached to that comment, feel free to remove it.